tests/MongoDB.Driver.Core.Tests/Core/Helpers/MockConnection.cs

```diff
@@ -31,21 +32,38 @@ public class MockConnection : IConnection
     {
         // fields
         private ConnectionId _connectionId;
+        private readonly ConnectionSettings _connectionSettings;
```

Comment:
added several changes to correspond to the particular parts of `BinaryConnection` (`closed/-ing` events and `expire` logic).

Reply:
Ack